crates/conjure_core/src/rules/rule_semantics.md

```diff
@@ -0,0 +1,16 @@
+## Priorities
+
+Rule are applied in priority order (highest number first).
+
+Rule priority levels are currently the following:
+
+| Priority level | Usage                                                                                     |
+| ---            | -----                                                                                     | 
+| **9001**   | **Total evaluation**                                                                          |
```

Comment:
if you are defining these, and re-ordering them, maybe rather than having a list of 'magic numbers', there could be some simple macros, like (feel free to rename) partialeval!(2) would map to 9002, which will (hopefully make these easier to read, and also mean it's easier to rearrange later if required?

Reply:
I agree; however, I couldn't find a way to make this work nicely. The underlying rule-set macro takes in numeric literals, not expressions, so a macro would require leading zeros. For example, partialeval!(002) to mean 900.